API/fleece/Fleece.hh

```diff
@@ -345,8 +345,11 @@ namespace fleece {
             FLTrust trust =kFLUntrusted,
             SharedKeys sk =nullptr,
             slice externDestination =nullslice) noexcept
-        :_doc(FLDoc_FromResultData(FLSliceResult(fleeceData), trust, sk, externDestination))
-        { }
+        {
+            FLSliceResult res = FLSliceResult(fleeceData);
```

Comment:
Good catch! I'd forgotten that the alloc_slice to FLSliceResult conversion is "magic" this way. I'd rather fix it by just avoiding the magic conversion, i.e.

`:_doc(FLDoc_FromResultData({fleeceData.buf, fleeceData.size}), trust, sk, externDestination))`

Reply:
done